File: radar-web-app/src/app/entities/infraction.ts
import {Vehicule} from "./vehicule";
import {Radar} from "./radar";
import {Proprietaire} from "./proprietaire";

export interface Infraction{

  id?: number ;
  date?:Date;
  vitesseVehicule?:number;
  montant?:number;
  vehicule?:Vehicule;
  radar?:Radar;

}

File: infraction-service/src/main/java/ma/enset/infractionservice/entities/Infraction.java
package ma.enset.infractionservice.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;


@Entity
@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class Infraction {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long numeroRadar;
    private Date date;
    private String matricule;
    private Float vitesseVehicule;
    private Float vitesseMax;
    private Float montant;
    @Transient
    private Vehicule vehicule;
    @Transient
    private Radar radar;


}
